### DISCUSSION TITLE
Updating Keycloak to 24.x?

### DISCUSSION THREAD

**Discussion by famod** [quarkusio/quarkus#39888] [Community]
Hi all (especially @sberyozkin),

are there any plans for updating Keycloak from 23.0.7 to 24.0.2?

**Comment by sberyozkin** [comment]
Hi @famod, interesting that you starting this discussion and me opening #39889 happened exactly at the same time =- 60 secs :-)

So yes, once we figure out with Pedro how to avoid updating most of the tests not having to enter the 3 extra fields then it will be ready to go

**Reply by famod** [reply]
Ha, I even checked PRs before starting this discussion. :laughing: 

Very nice. Is that going to end up only in Quarkus 3.10 (or even 3.11)? I'm on 3.9 currently and I'm wondering about compatibility.

**Reply by sberyozkin** [reply]
@famod I'm not sure right now, if, for the tests to pass users would need to update the tests to pass the extra properties then it would be tricky to backport. May be we'll update DevServices for Keycloak to register those properties automatically for the default realm.

As far as the compatibility is concerned, are you using `quarkus-keycloak-authorization` or `quarkus-keycloak-admin-client` ? In that case it would need to be checked, I'm not planning to ask for a backport to 3.9, with the major KC version update, but please propose the backport if you'd like

**Reply by famod** [reply]
We're using `quarkus-keycloak-admin-client` and it seems Quarkus 3.9.2 doesn't have an issue with KC 24 (at least for our setup and use cases).

**Reply by sberyozkin** [reply]
Thanks @famod, it looks unlikely though that we will be able to backport, due to impact on the tests. I'm not 100% sure it will make it to 3.10
